### DISCUSSION TITLE
Change/Replace R2 Metric ?

### DISCUSSION THREAD

**Discussion by daniel-caichac-DHI** [DHI/modelskill#234] [General]
Hi, I am wondering wether we should change the R2 metric. Let me explain myself.
Here I have some data. The regression does not look too bad yet R2 is negative
![image](https://github.com/DHI/modelskill/assets/97288080/15361cea-e056-4c64-83b0-2e072d4b2f24)
Which, given its definition, its ok.  However, the most common or most known definition of R2 is the square of `R`, with R being the `R=Pearson correlation coefficient`. In this case for my data `R_pearson=0.84` and therefore `R2=0.7198`.
If I plot my data in excel and do a linear regression and click show R2 I get this indeed.
![image](https://github.com/DHI/modelskill/assets/97288080/2539471c-07ec-45f0-bbd1-d44c13a606c4)

Also, if I run the **Mike Zero Time Series comparison (™)** I get the same value (ie, Pearson R2)
![image](https://github.com/DHI/modelskill/assets/97288080/dcec0e31-4cbc-4a5c-b748-c2b9da30c4ea)

I am not saying that one is better than the other, but shouldn't we stick to the most common one? Otherwise we could be potentially adding a source of error.

@jsmariegaard / @ecomodeller

**Comment by ecomodeller** [comment]
Read more about this topic in https://scikit-learn.org/stable/modules/model_evaluation.html#explained-variance-score

One of the motivations behind having `modelskill` as an open-source library is the that the formulation of metrics can be verified, but we should strive to use standardized nomenclature if possible, to avoid confusion.

**Comment by ecomodeller** [comment]
In the case of negative correlation, e.g. a Pearson correlation coefficient of -1, by squaring it you end up with a score of 1.0. Rather difficult metric to interpret.

**Reply by daniel-caichac-DHI** [reply]
Ok I found this very illustrative website where they show r2 and R2 and they agree that the current implementation (as is now) is indeed better
https://towardsdatascience.com/r%C2%B2-or-r%C2%B2-when-to-use-what-4968eee68ed3#:~:text=The%20Pearson%20correlation%20coefficient%20(r,the%20strength%20of%20a%20model.

but it is annoying that most software use the other one (person R square), include Mike Zero Time Series Comparison
